shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java

```diff
@@ -668,16 +666,41 @@ public long configureForTextureLayerComposition(
     return textureId;
   }
 
+  /**
+   * Translates an original touch event to have the same locations as the ones that Flutter
+   * calculates (because original + flutter's - original = flutter's).
+   *
+   * @param originalEvent The saved original input event.
+   * @param pointerCoords The coordinates that Flutter thinks the touch is happening at.
+   */
+  private void translateNonVirtualDisplayMotionEvent(
+      MotionEvent originalEvent, PointerCoords[] pointerCoords) {
+    if (pointerCoords.length < 1) {
+      return;
+    }
+
+    float xOffset = pointerCoords[0].x - originalEvent.getX();
+    float yOffset = pointerCoords[0].y - originalEvent.getY();
+
+    originalEvent.offsetLocation(xOffset, yOffset);
+  }
+
   @VisibleForTesting
   public MotionEvent toMotionEvent(
       float density, PlatformViewsChannel.PlatformViewTouch touch, boolean usingVirtualDiplay) {
     MotionEventTracker.MotionEventId motionEventId =
         MotionEventTracker.MotionEventId.from(touch.motionEventId);
     MotionEvent trackedEvent = motionEventTracker.pop(motionEventId);
 
+    // Compute this early so it can be used as input to translateNonVirtualDisplayMotionEvent.
+    PointerCoords[] pointerCoords =
+        parsePointerCoordsList(touch.rawPointerCoords, density)
+            .toArray(new PointerCoords[touch.pointerCount]);
+
     if (!usingVirtualDiplay && trackedEvent != null) {
-      // We have the original event, deliver it as it will pass the verifiable
+      // We have the original event, deliver it after offsetting as it will pass the verifiable
       // input check.
+      translateNonVirtualDisplayMotionEvent(trackedEvent, pointerCoords);
```

Comment:
after this call, can we add something like the following:

Can you do something here:

if (BuildConfig.DEBUG) {
  if (InputManager..verifyInputEvent(trackedEvent) == null) {
    throw new Exception(...)
  }
}

that will act like as a booby-trap if something ever goes wrong

Reply:
Added - I also added the inputmanager as a member of the class, so we didn't have to get it from the context in each invocation of `toMotionEvent`. But I can change it to do that if you prefer.

Reply:
👍